modules/integration/tests-ui-integration/src/test/java/org/wso2/carbon/mb/ui/test/dlc/DLCDurableTopicTestCase.java

```diff
@@ -0,0 +1,284 @@
+     *
+     * @throws Exception
+     */
+    @BeforeMethod()
+    public void cleanDeadLetterChannel() throws Exception {
+        driver.get(getLoginURL());
+        LoginPage loginPage = new LoginPage(driver);
+        homePage = loginPage.loginAs(mbServer.getContextTenant()
+                                             .getContextUser().getUserName(), mbServer
+                                             .getContextTenant().getContextUser()
+                                             .getPassword());
+
+        DLCBrowsePage dlcBrowsePage = homePage.getDLCBrowsePage();
+        //Testing delete messages
+        DLCContentPage dlcContentPage = dlcBrowsePage.getDLCContent();
+        dlcContentPage.deleteAllDLCMessages();
+    }
+
+    /**
+     * This method will add durable topic messages to dead letter channel.
+     *
+     * @throws AndesClientConfigurationException
+     * @throws NamingException
+     * @throws JMSException
+     * @throws IOException
+     * @throws AndesClientException
+     */
+    @BeforeMethod(dependsOnMethods = {"cleanDeadLetterChannel"})
+    public void addDurableTopicMessagesToDLC()
+            throws AndesClientConfigurationException, NamingException,
+                   JMSException, IOException, AndesClientException {
+        // Get current "AndesAckWaitTimeOut" system property.
+        defaultAndesAckWaitTimeOut = System.getProperty(AndesClientConstants.
+                                                                ANDES_ACK_WAIT_TIMEOUT_PROPERTY);
+
+        // Setting system property "AndesAckWaitTimeOut" for andes
+        System.setProperty(AndesClientConstants.ANDES_ACK_WAIT_TIMEOUT_PROPERTY, "0");
+
+        // Creating a initial JMS consumer client configuration
+        AndesJMSConsumerClientConfiguration consumerConfig = new
+                AndesJMSConsumerClientConfiguration(ExchangeType.TOPIC, DLC_TEST_DURABLE_TOPIC);
+        // Amount of message to receive
+        consumerConfig.setDurable(true, DLC_TEST_DURABLE_TOPIC);
+        consumerConfig.setSubscriptionID("durable-topic-sub-1");
+        consumerConfig.setMaximumMessagesToReceived(EXPECTED_COUNT + 200L);
+        consumerConfig.setAcknowledgeMode(JMSAcknowledgeMode.CLIENT_ACKNOWLEDGE);
+        consumerConfig.setAcknowledgeAfterEachMessageCount(EXPECTED_COUNT + 200L);
+
+        AndesJMSPublisherClientConfiguration publisherConfig =
+                new AndesJMSPublisherClientConfiguration(ExchangeType.TOPIC, DLC_TEST_DURABLE_TOPIC);
+        publisherConfig.setNumberOfMessagesToSend(SEND_COUNT);
+
+        consumerClient = new AndesClient(consumerConfig, true);
+        consumerClient.startClient();
+
+        AndesClient publisherClient = new AndesClient(publisherConfig, true);
+        publisherClient.startClient();
+
+        waitUntilMessagesReceived(6L);
+
+        //Thread sleep until messages sent to DLC after breaching maximum number of retrying
+        AndesClientUtils.sleepForInterval(80000L);
```

Comment:
Can we re-use these sleep intervals ? is it different based on the amount of messages ?